src/mower_logic/src/mower_logic/behaviors/DockingBehavior.cpp

```diff
@@ -47,8 +59,9 @@ bool DockingBehavior::approach_docking_point() {
     mbf_msgs::MoveBaseGoal moveBaseGoal;
     moveBaseGoal.target_pose = docking_approach_point;
     moveBaseGoal.controller = "FTCPlanner";
-    auto result = mbfClient->sendGoalAndWait(moveBaseGoal);
-    if (result.state_ != result.SUCCEEDED) {
+
+    auto result = waitForResultOrAborted(mbfClient, moveBaseGoal);
+    if (aborted || result.state_ != actionlib::SimpleClientGoalState::SUCCEEDED) {
```

Comment:
Couldn't the `if (...)` stay as it was? I really hope that the state can't be `SUCCEEDED` after `cancelGoal()` has been called.

Reply:
I don't see much of a benefit for removing this if check, in comparison to the work of understanding how cancellation is handled, and making sure that it won't change in the future.

Reply:
Alright, fine for me. `result.state_ != result.SUCCEEDED` no longer works? I would have expected same that `result` is still the same type as before.

Reply:
Done.